### ISSUE TITLE
Director::alternate_base_url breaks `Session` cookie path for CMS

### ISSUE THREAD

**Issue by dhensby** [silverstripe/silverstripe-framework#5109]
Steps to replicate:

Set alternate_base_url to an absolute domain, eg: `Config::inst()->update('Director', 'alternate_base_url', 'http://www.example.com/');`

Go to `example.com/admin`, login with valid credentials.

Notice infinite redirection loop of /Security/Login?BackURL=/admin/pages <-> /admin/pages

This is because `Session::inst_start()` looks at the `Director::baseURL()` if no `cookie_path` is explicitly set (see https://github.com/silverstripe/silverstripe-framework/blob/7312692a09ad32c7aa349a43d9aad00b0a353d21/control/Session.php#L360)

**Comment by tractorcow** [comment]
There doesn't seem to be a `Director::relativeBaseURL()`; I think we need to extract the root-relative url from the base in this case, but it'll need a new method I think. E.g. One that converts `http://www.example.com/` to `/`.

**Comment by dhensby** [comment]
indeed - seems to be a misunderstanding of what URL means ;)

Basically, base_url seems to be interchangable between a URL and a subdir

**Comment by tractorcow** [comment]
Yeah well, we need a subdir, not a url, so the method call is bad. :D

**Comment by whereisaaron** [comment]
Just wasted a bunch of time hitting and tracing this bug too. Setting an explicit `cookie_path` was a workaround to dodge the bad code.

There is `Director:makeRelative($url)` which might work to extract a sensible value for the cookie path?

**Comment by maxime-rainville** [comment]
I tried replicating this in SS4, but didn't manage to. But looking at the code, it should still have the same problem.
